### PR TITLE
[M876] Fish and Benthic ingest commands

### PR DESCRIPTION
**Pull request by gridcell**

### CODE REVIEW

**Comment by gridcell** [comment]
[add caribbean benthic and fish attributes; update 'region' for existing attributes](https://trello.com/c/Y7FZUEZK/876-add-caribbean-benthic-and-fish-attributes-update-region-for-existing-attributes)

File: src/tools/management/commands/fish_update_working2.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
+import argparse
```
Comment (saanobhaai) [inline src/tools/management/commands/fish_update_working2.py]: maybe we should add a comment at the top of this file and the benthic equivalent warning user to consider using the ingester-based commands instead

File: src/api/utils/castutils.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from decimal import ROUND_HALF_DOWN, Decimal, localcontext
+
+
+def to_decimal(val, max_digits, precision):
+    if str(val) == "" or val is None:
+        return None
+
+    with localcontext() as ctx:
```
[the remark this answers is not in the document]
Comment (gridcell) [reply src/api/utils/castutils.py]: partially yeah

File: src/tools/management/commands/fish_update_working.py [hunk 3]
```diff
@@ -49,6 +49,10 @@ def __init__(self):
         self.mode = 'f'
 
     def add_arguments(self, parser):
+        # f = family
```
[the remark this answers is not in the document]
Comment (gridcell) [reply src/tools/management/commands/fish_update_working.py]: removed fish_update_working2 as well

File: src/tools/management/commands/fish_update_working2.py (hunk 1 above)
Comment (gridcell) [reply src/tools/management/commands/fish_update_working2.py]: I removed the file

File: src/api/ingest/attributes_ingester.py [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+
+    def ingest(self, dry_run=False):
+        self.log = []
+        csvreader = csv.DictReader(self._file, delimiter=",")
+        n = 2
+        is_successful = True
+        for row in csvreader:
+            try:
+                with transaction.atomic():
+                    sid = transaction.savepoint()
```
[the remark this answers is not in the document]
Comment (gridcell) [reply src/api/ingest/attributes_ingester.py]: can't get around this

**Pull request closed without merging**